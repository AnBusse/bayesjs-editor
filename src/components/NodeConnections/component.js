import { addIndex, map } from 'ramda';

import NodeConnection from 'components/NodeConnection';
import NodeDivider from 'components/NodeDivider';
import PropTypes from 'prop-types';
import React from 'react';
import SvgText from 'components/SvgText';
import { sizePropTypes } from 'models';

const mapIndexed = addIndex(map);
const mapConnectionsCreator = cy => mapIndexed(({ networkName, ...props }, index) => (
  <NodeConnection
    key={networkName}
    name={networkName}
    cy={cy}
    index={index}
    {...props}
  />
));

const NodeConnections = ({
  connections,
  dividerY,
  textX,
  textY,
  size,
  circlesY,
  textHeight,
}) => {
  const mapConnections = mapConnectionsCreator(circlesY);

  return (
    <g>
      <NodeDivider y={dividerY} width={size.width} />
      <SvgText text="UniÃµes:" x={textX} y={textY} width={51} height={textHeight} />
      {mapConnections(connections)}
    </g>
  );
};

NodeConnections.propTypes = {
  // eslint-disable-next-line react/forbid-prop-types
  connections: PropTypes.array.isRequired,
  dividerY: PropTypes.number.isRequired,
  textX: PropTypes.number.isRequired,
  textY: PropTypes.number.isRequired,
  size: sizePropTypes.isRequired,
  circlesY: PropTypes.number.isRequired,
  textHeight: PropTypes.number.isRequired,
};

export default (NodeConnections);
