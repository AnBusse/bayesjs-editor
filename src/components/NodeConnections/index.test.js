import React from 'react';
import { shallow } from 'enzyme';
import NodeConnections from './component';

const defaultProps = {
  connections: [{
    networkId: 'id-1',
    networkName: 'network1',
    color: '#ddd',
    nodeId: 'node-id-1',
  }, {
    networkId: 'id-2',
    networkName: 'network2',
    color: '#ddd',
    nodeId: 'node-id-2',
  }],
  dividerY: 61,
  textX: 5,
  textY: 61,
  size: { height: 100, width: 160 },
  circlesY: 71,
  textHeight: 20,
};

const shallowComponent = (props = {}) => {
  const compProps = { ...defaultProps, ...props };

  return shallow(<NodeConnections {...compProps} />);
};

describe('NodeConnections Component', () => {
  it('matches snapshot', () => {
    const component = shallowComponent();

    expect(component).toMatchSnapshot();
  });
});
