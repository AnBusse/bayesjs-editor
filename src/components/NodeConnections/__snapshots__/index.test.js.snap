// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NodeConnections Component matches snapshot 1`] = `
<g>
  <NodeDivider
    width={160}
    y={61}
  />
  <SvgText
    height={20}
    text="UniÃµes:"
    width={51}
    x={5}
    y={61}
  />
  <NodeConnection
    color="#ddd"
    cy={71}
    index={0}
    key="network1"
    name="network1"
    networkId="id-1"
    nodeId="node-id-1"
  />
  <NodeConnection
    color="#ddd"
    cy={71}
    index={1}
    key="network2"
    name="network2"
    networkId="id-2"
    nodeId="node-id-2"
  />
</g>
`;
